sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptor/LoggingInterceptor.java

```diff
@@ -0,0 +1,327 @@
+        if (!logDetailLevel.shouldLogHeaders()) {
+            return;
+        }
+
+        Set<String> headerNames = headers.names();
+
+        for (String headerName : headerNames) {
+            sb.append(headerName).append(":");
+
+            if (allowedHeaderNames.contains(headerName.toLowerCase(Locale.ROOT))) {
+                List<String> headerValues = headers.values(headerName);
+
+                for (String headerValue : headerValues) {
+                    sb.append(headerValue);
+                    sb.append(System.lineSeparator());
+                }
+            } else {
+                sb.append(REDACTED_PLACEHOLDER);
+                sb.append(System.lineSeparator());
+            }
+        }
+    }
+
+    /**
+     * Attempts to retrieve and parse the Content-Length header into a numeric representation.
+     *
+     * @param logger  Logger used to log a warning if the Content-Length header is an invalid number.
+     * @param headers HTTP headers that are checked for containing Content-Length.
+     * @return long value indicating the content length of the Request or Response
+     */
+    private long getContentLength(ClientLogger logger, Headers headers) {
+        long contentLength = 0;
+        String contentLengthString = headers.get("Content-Length");
+
+        if (CoreUtils.isNullOrEmpty(contentLengthString)) {
+            return contentLength;
+        }
+
+        try {
+            contentLength = Long.parseLong(contentLengthString);
+        } catch (NumberFormatException | NullPointerException e) {
+            logger.warning("Could not parse the HTTP header content-length: '{}'.",
+                headers.get("content-length"), e);
+        }
+
+        return contentLength;
+    }
+
+    /**
+     * Determines if the request or response body should be logged.
+     *
+     * <p>The request or response body is logged if the Content-Type is not "application/octet-stream" and the body
+     * isn't empty and is less than 16KB in size.</p>
+     *
+     * @param contentTypeHeader Content-Type header value.
+     * @param contentLength     Content-Length header represented as a numeric.
+     * @return A flag indicating if the request or response body should be logged.
+     */
+    private boolean shouldBodyBeLogged(String contentTypeHeader, long contentLength) {
+        return !ContentType.APPLICATION_OCTET_STREAM.equalsIgnoreCase(contentTypeHeader)
+            && contentLength != 0
+            && contentLength < MAX_BODY_LOG_SIZE;
+    }
+
+    @NonNull
+    @Override
+    public Response intercept(@NonNull Chain chain) throws IOException {
+        Request request = chain.request();
+        Response response;
+
+        // No logging will be performed, trigger a no-op.
+        if (logDetailLevel == LogDetailLevel.NONE) {
+            return chain.proceed(request);
+        }
+
+        final ClientLogger logger = new ClientLogger(request.method() + " " + getRedactedUrl(request.url()));
```

Comment:
In the Java SDK a context parameter called "caller-method" is logged here. In absence of a context object, for now I'm logging the HTTP method type and the URL. Need to talk to someone on the java team to figure what this message looks like.